### DISCUSSION TITLE
Extract JSBSim's wind system as an actor for UE5.1

### DISCUSSION THREAD

**Discussion by junaidbinsarfraz** [JSBSim-Team/jsbsim#810] [Support requests]
I have an [Ultra Dynamic Sky](https://www.unrealengine.com/marketplace/en-US/product/6a1eafd3a6f34a929725585062b403a2) plugin for UE5.1 weather system (Blueprints). It has an all-weather system and wind to interact with the weather system. I can assign a _custom wind actor_ for the wind system.

In JSBSim, also has its own wind system which interacts with the flight (like DRAG, SIDE, and LIFT). If I use both wind systems then the result will be very ambiguous. **The best possible way is to attach JSBSim's wind actor with the plugin's wind actor** so that there will be no contradictions in the scene and physics. I am not sure how to attach JSBSim's wind to the plugin's wind actor. I am even not sure if it is a good idea or not.

Looking forward to ideas and demo implementations for this problem.

**Comment by seanmcleod** [comment]
If the Ultra Dynamic Sky plugin has some API where you get to specify the wind and/or an API for you to retrieve the current wind conditions then simply update the wind in JSBSim on each time step.

https://github.com/JSBSim-Team/jsbsim/blob/5ac73c2b555ac7b6b97a1ef43f3f47f1f68a6f0f/src/models/atmosphere/FGWinds.cpp#L497-L506
